### PR TITLE
Add git-ignore-io

### PR DESCRIPTION
**Pull request by Lee-W**
Add git ignore-io to generate sample gitignore file from gitignore.io
(Sorry, I accidentally closed previous pull requests)

### CODE REVIEW

File: bin/git-ignore-io [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+show_usage() {
+    echo "Usage:"
+    echo "  git ignore-io <types>...                    Show gitignore template"
+    echo "                [-a|--append] <types>...      Append new .gitignore content to .gitignore under the current directory"
+    echo "                [-e|--export] <types>...      Export new .gitignore to the current directory (The old one will be replaced)"
+    echo "                [-l|--list-in-table]          Print available types in table format"
+    echo "                [-L|--list-alphabetically]    Print available types in alphabetical order    "
+    echo "                [-s|--search] <word>          Search word in available types"
+    echo "                [-t|--show-update-time]       Show the last modified time of ~/.gi_list (where the list of available types stored)"
```
Comment (nicolaiskogheim) [inline bin/git-ignore-io]: > where the list of available types *is* stored

File: bin/git-ignore-io [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+check_list_exist
+if [[ $# -eq 0 ]]; then
+    show_usage
+else
+    case $1 in
+        -a|--append|-e|--export)
+            opt=$1
+            shift
+            if [[ $# -eq 0 ]]; then
+                echo "There should be at least one type!!!"
```
Comment (nicolaiskogheim) [inline bin/git-ignore-io]: I think you should remove the exclamation marks here.

File: man/git-ignore-io.1.html [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+<p>Get sample gitignore file from <a href="https://www.gitignore.io">gitignore.io</a></p>
+
+<h2 id="OPTIONS">OPTIONS</h2>
+
+<p>[-a|--append] <var>types</var>...      Append new .gitignore content to .gitignore under the current directory<br />
+[-e|--export] <var>types</var>...      Export new .gitignore to the current directory (The old one will be replaced)<br />
+[-l|--list-in-table]          Print available types in table format<br />
+[-L|--list-alphabetically]    Print available types in alphabetical order<br />
+[-s|--search] <var>word</var>          Search word in available types<br />
+[-t|--show-update-time]       Show the last modified time of ~/.gi_list (where the list of available types stored)<br />
```
Comment (nicolaiskogheim) [inline man/git-ignore-io.1.html]: > where the list of available types **is** stored

File: bin/git-ignore-io (hunk 2 above)
Comment (Lee-W) [inline bin/git-ignore-io]: Do you mean line 80?

Comment (nicolaiskogheim) [inline bin/git-ignore-io]: 117, sorry.

**Comment by nicolaiskogheim** [comment]
I think this will be really usefull to have, and I'm looking forward to if this is merged.

Can you take a look at the other man-pages and try to mimic them more? I want you to review the synopsis and options sections.

File: bin/git-ignore-io (hunk 2 above)
Comment (Lee-W) [inline bin/git-ignore-io]: I remove the one for search but reserver the one for append and export which I think is needed.

Comment (nicolaiskogheim) [inline bin/git-ignore-io]: I believe a `usage` message will suffice, but a warning is okay. But I only meant that you should remove the three exclamation marks.

**Comment by nicolaiskogheim** [comment]
> `-e` or `--export` to replace `.gitignore` with the result

Maybe this option should be `-r` and `--replace`. I don't think `export` really says what it does.

**Comment by Lee-W** [comment]
@nicolaiskogheim 
I'm working on the man-pages.
The `replace` option is great. I think `export` is weird, but cannot figure out a better name. Thanks!

**Comment by nicolaiskogheim** [comment]
How about changing `export` to `replace`?

**Comment by Lee-W** [comment]
That's great. I updated it.

**Comment by Lee-W** [comment]
@nicolaiskogheim I updated the manual.

**Comment by nicolaiskogheim** [comment]
:+1:

Question: Will the gitignore be placed in whatever directory is the current one when you run `git ignore-io`?

**Comment by Lee-W** [comment]
@nicolaiskogheim 
Sorry for my English. 
I'm not quite understand your question.
The `git ignore-io` will create or append content to `./.gitignore`.
Did I answer you question?

**Comment by nicolaiskogheim** [comment]
Hey! I'm so sorry, I forgot about this. I have been using your command the last month or so, and I find it really usefull.

I have tested this on OSX 10.9 with zsh, and FreeBSD. No problems here

**Comment by Lee-W** [comment]
Glad that you like this command~

**Comment by hemanth** [comment]
Let us merge this for now!

**Comment by raviranjananalist** [comment]
For Expert 5 Packers and Movers Hyderabad go to  
http://www.expert5th.in/packers-and-movers-hyderabad/
For Expert 5 Packers and Movers Bangalore go to 
http://www.expert5th.in/packers-and-movers-bangalore/
For Expert 5 Packers and Movers Mumbai go to 
http://www.expert5th.in/packers-and-movers-mumbai/

**Comment by raviranjananalist** [comment]
For Expert 5 Packers and Movers Gurgaon go to 
http://www.expert5th.in/packers-and-movers-gurgaon/
For Expert 5 Packers and Movers Delhi go to 
http://www.expert5th.in/packers-and-movers-delhi/
For Expert 5 Packers and Movers Pune go to  
http://www.expert5th.in/packers-and-movers-pune/

**Comment by raviranjananalist** [comment]
For Expert 5 Packers and Movers Chennai go to 
http://www.expert5th.in/packers-and-movers-chennai/
For Expert 5 Packers and Movers Noida go to 
http://www.expert5th.in/packers-and-movers-noida/
For Expert 5 Packers and Movers Thane go to 
http://www.expert5th.in/packers-and-movers-thane/
For Expert 5 Packers and Movers Navi Mumbai go to 
http://www.expert5th.in/packers-and-movers-navimumbai/
For Expert 5 Packers and Movers Ghaziabad go to 
http://www.expert5th.in/packers-and-movers-ghaziabad/
For Expert 5 Packers and Movers Faridabad go to 
http://www.expert5th.in/packers-and-movers-faridabad/

**Comment by raviranjananalist** [comment]
For Expert 5 Packers and Movers Bangalore go to 
http://www.expert5th.in/packers-and-movers-bangalore/
For Expert 5 Packers and Movers Pune go to  
http://www.expert5th.in/packers-and-movers-pune/
For Expert 5 Packers and Movers Mumbai go to 
http://www.expert5th.in/packers-and-movers-mumbai/

**Pull request merged**